tests/cxx/implicit_ctor-d1.h

```diff
@@ -21,15 +22,44 @@ int ImplicitCtorFn(IndirectWithImplicitCtor);
 // IWYU: IndirectWithImplicitCtor is...*implicit_ctor-i2.h.*for autocast
 int ImplicitCtorRefFn(const IndirectWithImplicitCtor&);
 
+// Autocast assurance for header-defined functions still makes sense as opposed
+// to function definitions in source files.
+// IWYU: IndirectWithImplicitCtor needs a declaration
+// IWYU: IndirectWithImplicitCtor is...*implicit_ctor-i2.h.*for autocast
+inline int InlineImplicitCtorRefFn(const IndirectWithImplicitCtor&) {
+  return 1;
+}
+
+int NoAutocastFn(
+    // A subtle c++ point: forward-declaring is ok for nonconst, because
+    // you can't do implicit conversion to a non-const reference
+    // (implicit conversion involves creating a temporary, which
+    // doesn't bind to non-const references).
+    // IWYU: IndirectWithImplicitCtor needs a declaration
+    IndirectWithImplicitCtor& nonconst,
+    // Forward-declaring is ok because ptrref is a const reference
+    // to a *pointer*.
+    // IWYU: IndirectWithImplicitCtor needs a declaration
+    IndirectWithImplicitCtor* const& ptrref,
+    // Test parameter type uses that do not require special handling for
```

Comment:
I meant to put this comment on the `NoAutocastFn` itself.

Reply:
Oh, I had already forgotten that the first two parameters don't require header inclusion for "autocast" too...